sympy/core/expr.py

```diff
@@ -781,7 +781,11 @@ def _eval_interval(self, x, a, b):
         else:
             A = self.subs(x, a)
             if A.has(S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
-                A = limit(self, x, a)
+                if a.__lt__(b):
```

Comment:
You need to guard against symbolic a < b. Not sure what to do in that case. Maybe just assume a < b by default?